### PR TITLE
DDF-1639 Added schema for result type mapping

### PR DESCRIPTION
**Pull request by EIrwin**
@kcwire is hero
@bdeining 
@wmcnalli

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/347)

### CODE REVIEW

**Comment by bdeining** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/347)_

File: catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        elementFormDefault="qualified" attributeFormDefault="unqualified"
+        version="1.0.0">
+
+	<!-- bring in other schemas -->
+    <xsd:import schemaLocation="metacard.xsd" namespace="mc"/>
+
+    <xsd:element name="result">
+		<xsd:complexType>
+			<xsd:sequence>
+				<xsd:element name="relevance" type="xsd:double" minOccurs="0"/>
```
Comment (kcwire) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: I would add <documentation/> or comments here describing what relevance and distance are meant to be.  Basically steal the comments from the Result interface. !fix

Comment (EIrwin) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: !fixed

File: catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        elementFormDefault="qualified" attributeFormDefault="unqualified"
+        version="1.0.0">
+
+	<!-- bring in other schemas -->
+    <xsd:import schemaLocation="metacard.xsd" namespace="mc"/>
+
+    <xsd:element name="result">
+		<xsd:complexType>
+			<xsd:sequence>
+				<xsd:element name="relevance" type="xsd:double" minOccurs="0">
```
Comment (pklinef) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: Can we use minInclusive of 0 and maxInclusive of 1?

File: catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+					<xsd:annotation>
+						<xsd:documentation>A "relevance" score (typical range of 0-1, values over 1 are those that have been boosted) is used to specify a sort policy based on relevance.
+						</xsd:documentation>
+					</xsd:annotation>
+				</xsd:element>
+				<xsd:element name="distance" type="xsd:double" minOccurs="0">
```
Comment (pklinef) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: We could have a minInclusive of 0 here too.  We could calculate a maxInclusive for this one as well.

File: catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+					<xsd:annotation>
+						<xsd:documentation>The "distance" is a representation of the distance in meters, null if not part of a search that was sorted by distance.
+						</xsd:documentation>
+					</xsd:annotation>
+				</xsd:element>
+				<xsd:element ref="mc:metacard">
```
Comment (pklinef) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: Shouldn't this be a minOccurs of 1 and maxOccurs of 1?  Metacard ID is a required attribute so there should always be at least one metacard attribute for a result.

File: catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd (hunk 2 above)
Comment (kcwire) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: !completed

File: catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd (hunk 3 above)
Comment (kcwire) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: !completed

File: catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd (hunk 4 above)
Comment (kcwire) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: !completed

**Comment by wmcnalli** [comment]
LGTM :+1:

File: catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+				</xsd:element>
+				<xsd:element name="relevance" minOccurs="0">
+					<xsd:simpleType>
+						<xsd:annotation>
+							<xsd:documentation>A "relevance" score (typical range of 0-1, values over 1 are those that have been boosted) is used to specify a sort policy based on relevance.
+							</xsd:documentation>
+						</xsd:annotation>
+						<xsd:restriction base="xsd:double">
+							<xsd:minInclusive value="0"/>
+							<xsd:maxInclusive value="1"/>
```
Comment (shaundmorris) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: Maybe I'm rusty on my schemas. In the documentation you specify the possibility of values over 1, however the maxInclusive is 1 so doesn't that mean the schema prohibits having a value over 1?

Comment (kcwire) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: Good point - I'll remove the maxInclusive.

Comment (kcwire) [inline catalog/transformer/catalog-transformer-xml-binding/src/main/resources/result.xsd]: !fixed

**Comment by kcwire** [comment]
Closing this PR and opening another with the latest updates.

**Pull request closed without merging**